### PR TITLE
apm: add transaction.profiler_stack_trace_ids field to traces datastream mapping

### PR DESCRIPTION
**Pull request by kyungeunni**
## Proposed commit message

Add `transaction.profiler_stack_trace_ids field` to traces datastream mapping

_It requires the **package-spec v3.1.0** which should be shipped with the **next release of elastic-package**(current release is v0.96.1)_

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Wait for the new release of `elastic-package` which includes package version 3.1.0

## How to test this PR locally

clone the [elastic-package](https://github.com/elastic/elastic-package), then run:
```sh
❯ make clean
...
❯ make build
...
❯ make install
```
This will move the binary built from the latest commit to `$GOBIN`.

Then go to the apm package directory and checkout to this branch, then run:
```sh
❯ elastic-package check
...
❯ elastic-package lint
``` 
It shouldn't print any error logs, and look something like this:
image

### Test with Registry
```
❯ elastic-package stack up -v -d --version 8.13.0-SNAPSHOT
# once the stack is up(registry must be running), install the package
# integrations/packages/apm
❯ elastic-package build
...
❯ elastic-package install
```

When installed, Go to Kibana, and navigate to `Stack Management > Index Management > Component` and search for `traces`. the component should show the added field with the counted_keyword type.

image

## Related issues

- Relates https://github.com/elastic/apm-data/pull/202 
- Relates https://github.com/elastic/apm-server/issues/12054
- Relates https://github.com/elastic/elasticsearch/pull/105223

### CODE REVIEW

**Comment by kyungeunni** [comment]
Failure is expected until we have the newer version of `elastic-package` as the current version doesn't recognise `counted_keyword` type.

File: packages/apm/changelog.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 - version: 8.13.0-preview-1
```
Comment (carsonip) [inline packages/apm/changelog.yml:1]: needs a version bump > `8.13.0-preview-1705349440` to trigger a build

File: packages/apm/manifest.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-format_version: 3.0.1
+format_version: 3.1.0
 name: apm
 title: Elastic APM
 version: 8.13.0-preview-1
```
Comment (carsonip) [inline packages/apm/manifest.yml:4]: needs a version bump > `8.13.0-preview-1705349440` to trigger a build

Comment (kyungeunni) [reply packages/apm/manifest.yml]: Where's `1705349441` coming from?

**Review by inge4pres — APPROVED** [review]
🚀

**Pull request merged**